server/controllers/list.js

```diff
@@ -263,7 +284,53 @@ const updateListItem = (req, resp) => {
       const item = doc.items[itemIndex];
 
       doc
-        ? resp.status(200).json(item)
+        ? resp.status(200).json(itemToSend(item, userId))
+        : resp.status(404).send({ message: 'List data not found.' });
+    }
+  );
+};
+
+const voteForItem = (req, resp) => {
+  const { itemId, didCurrentUserVoted } = req.body;
+  const { id: listId } = req.params;
+  const {
+    user: { _id: userId }
+  } = req;
+
+  const dataToUpdate = didCurrentUserVoted
+    ? {
+        $pull: { 'items.$.voterIds': req.user._id },
+        $inc: { 'items.$.votesCount': -1 }
+      }
+    : {
+        $push: { 'items.$.voterIds': req.user._id },
+        $inc: { 'items.$.votesCount': 1 }
+      };
+
+  List.findOneAndUpdate(
+    {
+      _id: listId,
+      'items._id': itemId,
+      $or: [
+        { adminIds: req.user._id },
+        { ordererIds: req.user._id },
+        { purchaserIds: req.user._id }
+      ]
+    },
+    dataToUpdate,
+    { new: true },
+    (err, doc) => {
+      if (err) {
+        return resp.status(400).send({
+          message:
+            'An error occurred while updating the list data. Please try again.'
+        });
+      }
+      const itemIndex = doc.items.findIndex(item => item._id.equals(itemId));
+      const item = doc.items[itemIndex];
+
+      doc
+        ? resp.status(200).json(itemToSend(item, userId))
```

Comment:
Maybe change name `itemToSend` with `responseWithItem`? And if we would have such other methods then we should caal them same like 'responseWith...`?